slash-commands/slash-commands.js

```diff
@@ -115,14 +149,23 @@ CandyShop.SlashCommands = (function(self, Candy, $) {
 	self.part = function() {
 		Candy.Core.Action.Jabber.Room.Leave(self.currentRoom());
 	};
+	
+	/** Function: leave
+	 * Exits the current chat room
+	 *
+	 */
+	self.leave = function() {
+		Candy.Core.Action.Jabber.Room.Leave(self.currentRoom());
+	};
 
 	/** Function: topic
 	 * Sets the topic (subject) for the current chat room
 	 *
 	 * Parameters:
-	 * 	(String) topic The new topic for the room
+	 *    (String) topic The new topic for the room
 	 */
 	self.topic = function(topic) {
+		// TODO: .isModerator() && allowed checks
```

Comment:
Yes, \topic is configurable per-room feature, which I should be able to detect via... somthing. Still haven't had the time to figure out the mechanism yet, though if I recall correctly the same room data is sent whether you're a room moderator or not. This may only be the case for Prosody.

No check means there would mean the possibility silent failure. Users tend to think that means bugs, not intent.

Candy.View.Pane.Chat.onInfoMessage is what I've been using for all such messages, and is what the response would be in case of that otherwise silent failure.